src/components/Main.tsx

```diff
@@ -247,10 +251,17 @@ export const Main = () => {
     }, 0)
   }, [])
 
-  return (
+  // Wait for the sticky config to initialize before rendering anything
+  useAsyncEffect(async () => {
+    setLegacyLanding((await getStickyConfigValue('legacyLanding')) as boolean)
+  }, [])
+
+  return legacyLanding === undefined ? (
+    <LoadingScene />
+  ) : (
     <NavigationContainer theme={reactNavigationTheme}>
       <Stack.Navigator
-        initialRouteName={ENV.USE_WELCOME_SCREENS ? 'gettingStarted' : 'login'}
+        initialRouteName={ENV.USE_WELCOME_SCREENS && !legacyLanding ? 'gettingStarted' : 'login'}
```

Comment:
Don't you need to implement the same logic for `maybeHandleComplete` in `LoginScene.tsx`?